build/gulp/tasks/bundle.ts

```diff
@@ -2,70 +2,79 @@ import { task, series, parallel, src, dest } from 'gulp'
 import * as rimraf from 'rimraf'
 import * as webpack from 'webpack'
+import { argv } from 'yargs'
 
 import config from '../../../config'
+import sh from '../sh'
 
 const { paths } = config
 const g = require('gulp-load-plugins')()
 const { log, PluginError } = g.util
 
+const buildUMD = !!argv.umd
+const packageName: string = argv.package || 'react'
+
 // ----------------------------------------
 // Clean
 // ----------------------------------------
 
-task('clean:dist:es', cb => {
-  rimraf(`${config.paths.dist()}/es/*`, cb)
+task('bundle:clean:es', cb => {
+  rimraf(`${config.paths.packageDist(packageName)}/es/*`, cb)
 })
 
-task('clean:dist:commonjs', cb => {
-  rimraf(`${config.paths.dist()}/commonjs/*`, cb)
+task('bundle:clean:commonjs', cb => {
+  rimraf(`${config.paths.packageDist(packageName)}/commonjs/*`, cb)
 })
 
-task('clean:dist:umd', cb => {
-  rimraf(`${config.paths.dist()}/umd/*`, cb)
+task('bundle:clean:umd', cb => {
+  rimraf(`${config.paths.packageDist(packageName)}/umd/*`, cb)
 })
 
-task('clean:dist', parallel('clean:dist:es', 'clean:dist:commonjs', 'clean:dist:umd'))
+task('bundle:clean', parallel('bundle:clean:es', 'bundle:clean:commonjs', 'bundle:clean:umd'))
 
 // ----------------------------------------
 // Build
 // ----------------------------------------
-const componentsSrc = [paths.src('**/*.{ts,tsx}'), `!${paths.src('**/umd.ts')}`]
+const componentsSrc = [
+  paths.packageSrc(packageName, '**/*.{ts,tsx}'),
+  `!${paths.packageSrc(packageName, '**/umd.ts')}`,
+]
 
-task('build:dist:commonjs', () => {
+task('bundle:build:commonjs', () => {
   const tsConfig = paths.base('build/tsconfig.commonjs.json')
   const settings = { declaration: true }
-  const typescript = g.typescript.createProject(tsConfig, settings)
 
+  const typescript = g.typescript.createProject(tsConfig, settings)
   const { dts, js } = src(componentsSrc).pipe(typescript())
-  const types = src(paths.base('types/**'))
 
   return merge2([
-    dts.pipe(dest(paths.dist('commonjs'))),
-    js.pipe(dest(paths.dist('commonjs'))),
-    types.pipe(dest(paths.dist('types'))),
+    dts.pipe(dest(paths.packageDist(packageName, 'commonjs'))),
+    js.pipe(dest(paths.packageDist(packageName, 'commonjs'))),
   ])
 })
 
-task('build:dist:es', () => {
+task('bundle:build:es', () => {
   const tsConfig = paths.base('build/tsconfig.es.json')
   const settings = { declaration: true }
-  const typescript = g.typescript.createProject(tsConfig, settings)
 
+  const typescript = g.typescript.createProject(tsConfig, settings)
   const { dts, js } = src(componentsSrc).pipe(typescript())
-  const types = src(paths.base('types/**'))
 
   return merge2([
-    dts.pipe(dest(paths.dist('es'))),
-    js.pipe(dest(paths.dist('es'))),
-    types.pipe(dest(paths.dist('types'))),
+    dts.pipe(dest(paths.packageDist(packageName, 'es'))),
+    js.pipe(dest(paths.packageDist(packageName, 'es'))),
   ])
 })
 
-task('build:dist:umd', cb => {
+task('bundle:build:umd', cb => {
```

Comment:
actually, this might be very confusing if one would try to run

```
gulp bundle:build:umd
```

It seems odd that, for the sake of artefacts being generated, this command should be invoked with the following flag:

```
gulp bundle:build:umd --umd
```

Reply:
Will rework it to separate tasks:
- `gulp bundle --package react`
- `gulp bundle:with-umd --package react`